index.bs

```diff
@@ -1466,9 +1466,86 @@ these steps:
   1. [=/Resolve=] |p| with |result|.
 
 1. Return |p|.
+</div>
+
+## Drag and Drop ## {#drag-and-drop}
+
+<xmp class=idl>
+partial interface DataTransferItem {
+    Promise<FileSystemHandle?> getAsFileSystemHandle();
+};
+</xmp>
+
+During a <em>drag-and-drop operation</em>, dragged file and
+directory items are associated with [=file entries=] and [=directory entries=]
+respectively.
+
+<div class="note domintro">
+  : |handle| = await item . {{getAsFileSystemHandle()}}
+  :: Returns a {{FileSystemFileHandle}} object if the dragged item is file and a {{FileSystemDirectoryHandle}} object if the dragged item is a directory.
+</div>
+
+<div algorithm>
+
+The <dfn method for=DataTransferItem>getAsFileSystemHandle()</dfn> method steps are:
+
+1. Let |p| be [=a new promise=].
+
+1. Run the following steps [=in parallel=]:
 
+  1. If the {{DataTransferItem}} object is not in the <a spec=html>read/write
```

Comment:
I'm really sorry, I know we talked about this before, but thinking about this more, checking things like the read/write mode of the DataTransferItem has to be done before we run things in parallel.

So probably easiest to do it before you even create |p|, as first two steps of this overall algorithm ```1. If ...., return [=a promise resolved with=] `null`.```